torch/nn/utils/rnn.py

```diff
@@ -320,13 +321,22 @@ def pad_sequence(sequences, batch_first=False, padding_value=0):
         out_dims = (max_len, len(sequences)) + trailing_dims
 
     out_tensor = sequences[0].data.new(*out_dims).fill_(padding_value)
-    for i, tensor in enumerate(sequences):
-        length = tensor.size(0)
-        # use index notation to prevent duplicate references to the tensor
-        if batch_first:
-            out_tensor[i, :length, ...] = tensor
-        else:
-            out_tensor[:length, i, ...] = tensor
+    if padding_type == 'post':
+        for i, tensor in enumerate(sequences):
+            length = tensor.size(0)
+            if batch_first:
+                out_tensor[i, :length, ...] = tensor
+            else:
+                out_tensor[:length, i, ...] = tensor
+    elif padding_type == 'pre':
+        for i, tensor in enumerate(sequences):
+            length = tensor.size(0)
+            if batch_first:
+                out_tensor[i, -length:, ...] = tensor
+            else:
+                out_tensor[-length:, i, ...] = tensor
+    else:
+        raise ValueError("Padding_type must be 'post' or 'pre'")
```

Comment:
Nit: say what padding_type the user passed in:
`("pad_sequence: padding_type must be  'post' or 'pre', got: {}".format(padding_type)`